x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/NativeRolesStore.java

```diff
@@ -144,16 +144,19 @@ public void getRoleDescriptors(Set<String> names, final ActionListener<RoleRetri
     }
 
     public void deleteRole(final DeleteRoleRequest deleteRoleRequest, final ActionListener<Boolean> listener) {
-        securityIndex.prepareIndexIfNeededThenExecute(listener::onFailure, () -> {
-            DeleteRequest request = client.prepareDelete(SecurityIndexManager.SECURITY_INDEX_NAME,
+        if (securityIndex.isAvailable() == false) {
+            listener.onResponse(false);
```

Comment:
As above, I think `false` is incorrect if the shards were not available.
It's possible that the role will exist in a few seconds simply because the shards are available again.